client/client_example_test.go

```diff
@@ -94,5 +107,86 @@ func ExampleDynamicWatcher() {
 		panic(err)
 	}
 
+	// Run until the context is canceled.
+	<-ctx.Done()
+
+	// Output:
+}
+
+func ExampleNewControllerRuntimeSource() {
+	// Start a test Kubernetes API.
+	testEnv := envtest.Environment{}
+
+	k8sConfig, err := testEnv.Start()
+	if err != nil {
+		panic(err)
+	}
+
+	defer func() {
+		err := testEnv.Stop()
+		if err != nil {
+			klog.Error(err, "failed to stop the test Kubernetes API")
+		}
+	}()
+
+	// Create a context that can be explicitly canceled.
+	ctx, cancel := context.WithCancel(context.TODO())
+
+	dynamicWatcherReconciler, sourceChan := client.NewControllerRuntimeSource(1024)
+
+	// Create the dynamic watcher using the generated reconciler.
+	dynamicWatcher, err := client.New(k8sConfig, dynamicWatcherReconciler, nil)
+	if err != nil {
+		panic(err)
+	}
+
+	// Start the dynamic watcher in a separate goroutine to not block the main goroutine.
+	go func() {
+		err := dynamicWatcher.Start(ctx)
+		if err != nil {
+			panic(err)
+		}
+	}()
+
+	// Wait until the dynamic watcher has started.
+	<-dynamicWatcher.Started()
+
+	// Create a controller-runtime manager and register a simple controller.
+	options := ctrl.Options{
+		Namespace:              "default",
+		Scheme:                 scheme.Scheme,
+		MetricsBindAddress:     "0",
+		HealthProbeBindAddress: "0",
+		LeaderElection:         false,
+	}
+
+	mgr, err := ctrl.NewManager(k8sConfig, options)
+	if err != nil {
+		panic(err)
+	}
+
+	// This controller watches ConfigMaps and will reconcile any time the dynamic watcher sees a watched object is
+	// updated.
+	err = ctrl.NewControllerManagedBy(mgr).
+		For(&corev1.ConfigMap{}).
+		Watches(sourceChan, &handler.EnqueueRequestForObject{}).
+		Complete(&ctrlRuntimeReconciler{})
+
+	if err != nil {
+		panic(err)
+	}
+
+	// Simulate something canceling the context in 5 seconds so that the example exits.
+	go func() {
+		time.Sleep(5 * time.Second)
+
+		cancel()
+	}()
+
+	err = mgr.Start(ctx)
+	if err != nil {
+		panic(err)
+	}
+
 	// Output:
```

Comment:
There's no output since the fmt.Printf call is in a separate goroutine. I have to have this comment here or else Go won't run the example as a unit test, so it's essentially just running the function and validating that the stdout from it is empty.